.github/workflows/report_quiz/leaderboard/functions.sh

```diff
@@ -0,0 +1,9 @@
+# compute the user total score from more than one quiz_data
+compute_total_score(){
+    jq -s --argjson total_price 30000 'map((.correct_answers | map(.)) as $answers | (.user_answers // []) | map(.score=(.options_ids | map(select($answers[.] == "true")) | length)-(.options_ids | map(select($answers[.] == "false")) | length) | del(.options_ids))) | add | group_by(.first_name) | map((map(.score) | add) as $total_score | .[0] | del(.score) | .total_score=$total_score) | group_by(-.total_score) | to_entries | map(.price=$total_price/((.key+1)*2))' $@
```

Comment:
This is too compact, even tho it's understandable, make it digest please, not in a single line

Reply:
Yes, but the fact that it's a sequential instruction, don't affect his readability.

Reply:
Add in the comment that is up the method a simple use case représentation of what can be an input and an output, to help us see what the method is capable of if the JQ filter can not be on multi lines.

Reply:
This jq command reflects the function name and his comment.

Reply:
i agree with @Sanix-Darker , this is not easy to maintain. I am pretty sure you can split it into smaller sequential and logical chunks so that other people can understand it quicker. Performance is not an issue here, since this run in a runner on a cron, no one is waiting for this answer to be sent in less than 10 milliseconds. I am more for privileging readability/maintainability over speed here.

You also could write it in multiple lines for example, add some single line comments to your script, use re-usable variables etc which I think are supported by jq.

Reply:
@rakici @Sanix-Darker 
Is it this splitting enough?

```
map(
  (.correct_answers | map(.)) as $answers
  | (.user_answers // [])
  | map(
    .score=(
      .options_ids
      | map(select($answers[.] == "true"))
      | length
    ) - (
      .options_ids
      | map(select($answers[.] == "false"))
      | length
    )
    | del(.options_ids)
  )
)
| add
| group_by(.first_name)
| map(
  (map(.score) | add) as $total_score
  | .[0]
  | del(.score)
  | .total_score=$total_score
)
| group_by(-.total_score)
| to_entries
| map(.price=$total_price/((.key+1)*2))
```